### ISSUE TITLE
Declarative macro transcription does not take into account precedence

### ISSUE THREAD

**Issue by scrabsha** [ferrocene/specification#444]
Let's consider the following code:

```rust
macro_rules! two_times {
    ($e:expr) => {
        2 * $e
    };
}

fn main() {
    let a = two_times!(1 + 1);
    assert_eq!(a, 4);
}
```

Let's break down its expansion according to the current Ferrocene specs:
- `$e` matches the expression `1 + 1` ([20.4.1.2:6](https://spec.ferrocene.dev/macros.html#fls_k00bck2k8tde)),
- When transcribing, `$e` is replaced with `1 + 1` ([20.4.2:3](https://spec.ferrocene.dev/macros.html#fls_iw7322ycvhkc)),
- The final expansion is `2 * 1 + 1`.

Well, that's not what currently happens ([playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=e50f94203971423da7850dcbac7bc000)). `rustc -Zunpretty=expanded` even adds parenthesis around the `1 + 1`.

My understanding as that `expr` [metavariables](https://spec.ferrocene.dev/macros.html#metavariables) that match an [arithmetic expression](https://spec.ferrocene.dev/expressions.html#arithmetic-expressions) should have a higher [precedence](https://spec.ferrocene.dev/expressions.html#expression-precedence) than the rest of the macro substitution. As far as I can tell, this is not mentioned in Ferrocene.

Side note: I don't have much experience in how Ferrocene specs are structured, so I may have used the wrong word/missed something important. Sorry for that.

**Comment by Veykril** [comment]
Good catch we are not specifying none groups (aka invisible delimiters) at all currently which is in general a somewhat undesigned space in rust still https://github.com/rust-lang/rust/issues/67062. We should be specifying the grouping behavior for expressions at least though.
